sunpy/net/cdaweb/cdaweb.py

```diff
@@ -125,7 +125,14 @@ def fetch(self, query_results, *, path, downloader, **kwargs):
             fname = row['URL'].split('/')[-1]
             filepath = str(path).format(file=fname)
             # Manually cap max_splits at 3 to make CDAWeb happy
-            downloader.enqueue_file(row['URL'], filename=filepath, max_splits=3)
+            max_splits = 3
+            if "max_splits" in kwargs:
+                max_splits = kwargs.pop('max_splits')
+                max_splits = min(max_splits, 3)
+            # Unsupported keyword that is passed by the main fetch method
+            if "wait" in kwargs:
+                kwargs.pop("wait")
```

Comment:
This brings up an interesting point. I think we are missing some clarity on the `fetch()` API.

I don't think there is any documented expectation that all spare kwargs to `Fido.fetch()` get passed to `enqueue_file`? In the dev guide for Fido it says:

> `**kwargs` It is very important that fetch() methods accept extra keyword arguments that they don’t use, as the user might be passing them to other clients via Fido.

and while we don't exactly encourage it, I have seen clients implement custom kwargs in their fetch method, which according to that line should be supported, and this PR would break, or is already broken in other clients.

We should figure this mess out. What triggered you opening this PR in the first place?

Reply:
The error I saw on sunkit-image.

Reply:
> The error I saw on sunkit-image.

Link me?

Reply:
Doesn't exist anymore.

Reply:
https://dev.azure.com/sunpy/sunkit-image/_build/results?buildId=24056&view=logs&j=090c8010-07b3-5084-af5f-2960d012d632&t=15d054e4-a436-5390-d544-f8af6232a527